pkg/diagnostics/cluster/aggregated_logging/diagnostic.go

```diff
@@ -0,0 +1,173 @@
+)
+
+// AggregatedLogging is a Diagnostic to check the configurations
+// and general integration of the OpenShift stack
+// for aggregating container logs
+// https://github.com/openshift/origin-aggregated-logging
+type AggregatedLogging struct {
+	masterConfig     *configapi.MasterConfig
+	MasterConfigFile string
+	OsClient         *client.Client
+	KubeClient       *kclient.Client
+	result           types.DiagnosticResult
+}
+
+const (
+	AggregatedLoggingName = "AggregatedLogging"
+
+	loggingInfraKey = "logging-infra"
+	componentKey    = "component"
+	providerKey     = "provider"
+	openshiftValue  = "openshift"
+
+	fluentdServiceAccountName = "aggregated-logging-fluentd"
+)
+
+var loggingSelector = labels.Set{loggingInfraKey: "support"}
+
+//NewAggregatedLogging returns the AggregatedLogging Diagnostic
+func NewAggregatedLogging(masterConfigFile string, kclient *kclient.Client, osclient *client.Client) *AggregatedLogging {
+	return &AggregatedLogging{nil, masterConfigFile, osclient, kclient, types.NewDiagnosticResult(AggregatedLoggingName)}
+}
+
+func (d *AggregatedLogging) getScc(name string) (*kapi.SecurityContextConstraints, error) {
+	return d.KubeClient.SecurityContextConstraints().Get(name)
+}
+
+func (d *AggregatedLogging) getClusterRoleBinding(name string) (*authapi.ClusterRoleBinding, error) {
+	return d.OsClient.ClusterRoleBindings().Get(name)
+}
+
+func (d *AggregatedLogging) routes(project string, options kapi.ListOptions) (*routesapi.RouteList, error) {
+	return d.OsClient.Routes(project).List(options)
+}
+
+func (d *AggregatedLogging) serviceAccounts(project string, options kapi.ListOptions) (*kapi.ServiceAccountList, error) {
+	return d.KubeClient.ServiceAccounts(project).List(options)
+}
+
+func (d *AggregatedLogging) services(project string, options kapi.ListOptions) (*kapi.ServiceList, error) {
+	return d.KubeClient.Services(project).List(options)
+}
+
+func (d *AggregatedLogging) endpointsForService(project string, service string) (*kapi.Endpoints, error) {
+	return d.KubeClient.Endpoints(project).Get(service)
+}
+
+func (d *AggregatedLogging) daemonsets(project string, options kapi.ListOptions) (*kapisext.DaemonSetList, error) {
+	return d.KubeClient.DaemonSets(project).List(kapi.ListOptions{LabelSelector: loggingInfraFluentdSelector.AsSelector()})
+}
+
+func (d *AggregatedLogging) nodes(options kapi.ListOptions) (*kapi.NodeList, error) {
+	return d.KubeClient.Nodes().List(kapi.ListOptions{})
+}
+
+func (d *AggregatedLogging) pods(project string, options kapi.ListOptions) (*kapi.PodList, error) {
+	return d.KubeClient.Pods(project).List(options)
+}
+func (d *AggregatedLogging) deploymentconfigs(project string, options kapi.ListOptions) (*deployapi.DeploymentConfigList, error) {
+	return d.OsClient.DeploymentConfigs(project).List(options)
+}
+
+func (d *AggregatedLogging) Info(id string, message string) {
+	d.result.Info(id, message)
+}
+
+func (d *AggregatedLogging) Error(id string, err error, message string) {
```

Comment:
I think the tests have value.  I am leaning towards leaving it as is and to modify it if we find it becomes an issue going forward.